src/data/viteComboFetcher.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * **WARNING** This module is expected to be invoked from a Vite Plugin!
+ * 
+ * Aggregated all YAML Files in the ./combos Directory and interpretes 
+ */
+
+import { parse } from "yaml"
+import { DbCombosTableZod, DbTricksTableZod } from "../lib/database/schemas/CurrentVersionSchema"
+import { readFile } from "node:fs/promises"
+import { ZodError, z } from "zod"
+import {globby} from "globby"
+import { join } from "node:path"
+import { fileURLToPath } from "node:url"
+import { YamlComboTableSchemaZod } from "./viteSchemaGenerator"
+
+async function fetchYamlFile(path: string) {
+  const asObject = YamlComboTableSchemaZod.parse(parse(await readFile(path, "utf8")))
+
+  if(!asObject.tricks) {
+    throw new Error("Combos must contain at least one trick.")
+  }
+
+  asObject.tricks = asObject.tricks.map( trickId => ([trickId, "official"])) as any
+
+  // Makes sure the YAML has the right structure. 
+  return DbCombosTableZod.parse({...asObject, comboStatus: "official"})
+}
+
+/**
+ * This takes a list of Database Entities and creates a mapping of ID → List of Objects
+ * 
+ * This can be used to detect duplicates or to check if an Entity with a given ID exists.
+ */
+function createRecordLookup<T extends z.infer<typeof DbCombosTableZod> | z.infer<typeof DbTricksTableZod>>(allEntities: T[]) {
+  let idToEntityLookup: Record<number, T[]> = {};
+
+  allEntities.forEach(databaseEntity => {
+    if (!idToEntityLookup[databaseEntity.id]) {
+      idToEntityLookup[databaseEntity.id] = [];
+    }
+      idToEntityLookup[databaseEntity.id].push(databaseEntity);
+    });
+  return idToEntityLookup;
+}
+
+function findDuplicateKeys(allCombos: z.infer<typeof DbCombosTableZod>[]) {
+
+  let idToTrickLookup: Record<number, (z.infer<typeof DbCombosTableZod>)[]> = createRecordLookup(allCombos);
+
+  return Object.entries(idToTrickLookup).filter(([_,trick]) => trick.length > 1).map(([k,v]) => ({
+    id: k,
+    technicalNames: v.map( trick => trick.technicalName)
+  }))
+}
+
+function findUndefinedTrickReferences(allCombos: z.infer<typeof DbCombosTableZod>[], allTricks: z.infer<typeof DbTricksTableZod>[]) {
+  
+  let idToTrickLookup: Record<number, (z.infer<typeof DbTricksTableZod>)[]> = createRecordLookup(allTricks);
+
+  // issues define the ID of the Combo and an issue message
+  const issues: [number, string][] = []
+
+  allCombos.forEach( combo => {
+    if(combo.tricks) {
```

Comment:
Maybe there should also be an issue, when there is no (or only one) trick in a combo? 'All combos should have at least two tricks'

Reply:
Nevermind, I just realized this is done in `viteGetAllCombos`